### PR TITLE
Bug 1716426 : Fix typo in admission pluginConfig

### PR DESCRIPTION
**Pull request by mfojtik**
Fix: https://bugzilla.redhat.com/show_bug.cgi?id=1716426

### CODE REVIEW

File: bindata/v4.1.0/kube-apiserver/defaultconfig.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 apiVersion: kubecontrolplane.config.openshift.io/v1
 kind: KubeAPIServerConfig
 admission:
-  pluginConfigs:
+  pluginConfig:
```
Comment (danwinship) [inline bindata/v4.1.0/kube-apiserver/defaultconfig.yaml]: looking further down:

    network.openshift.io/RestrictedEndpointsAdmission:
      configuration:
        apiVersion: network.openshift.io/v1
        kind: RestrictedEndpointsAdmissionConfig
        restrictedCIDRs:
        - 10.3.0.0/16 # ServiceCIDR
        - 10.2.0.0/16 # ClusterCIDR

neither of those CIDRs is correct. I think the values here will just get overwritten at runtime with the correct values read from the network config status, but it's weird to have explicitly wrong values in there by default...

Comment (mfojtik) [reply bindata/v4.1.0/kube-apiserver/defaultconfig.yaml]: @danwinship what are the correct defaults?

Comment (mfojtik) [reply bindata/v4.1.0/kube-apiserver/defaultconfig.yaml]: we will remove this config and wait for network operator to set it.

Comment (deads2k) [reply bindata/v4.1.0/kube-apiserver/defaultconfig.yaml]: To be clear, he's talking about this code here: https://github.com/openshift/cluster-kube-apiserver-operator/tree/master/pkg/operator/configobservation/network

Comment (danwinship) [reply bindata/v4.1.0/kube-apiserver/defaultconfig.yaml]: `172.30.0.0/16` for ServiceCIDR and `10.128.0.0/14` for ClusterCIDR, but they can be overridden from install-config.yaml

Comment (deads2k) [reply bindata/v4.1.0/kube-apiserver/defaultconfig.yaml]: @mfojtik is that code typo'd too? https://github.com/openshift/cluster-kube-apiserver-operator/blob/master/pkg/operator/configobservation/network/observe_network.go#L18

Comment (mfojtik) [reply bindata/v4.1.0/kube-apiserver/defaultconfig.yaml]: yeah, that looks bugged as well... fixed.

**Comment by mfojtik** [comment]
@danwinship @deads2k updated

**Comment by squeed** [comment]
Not there yet:
`F0603 19:36:25.026838       1 cmd.go:66] failed to initialize admission: couldn't init admission plugin "network.openshift.io/RestrictedEndpointsAdmission": error reading config: Object 'Kind' is missing in '{"restrictedCIDRs":["10.128.0.0/14","172.30.0.0/16"]}'`

**Comment by deads2k** [comment]
/lgtm
/hold

holding so it doesn't cycle over and over.

@squeed for the future, have a look at the aos-devel thread with Derek and myself called "admission plugins and you" for a list of ownership and how we're handling integration for kube-apiserver and openshift-apiserver configuration in support of various features.  Everyone has a pet setting and we're looking for feature owners to own enabling their own features.

**Comment by mfojtik** [comment]
Replaced by https://github.com/openshift/cluster-kube-apiserver-operator/pull/494

**Pull request closed without merging**